include/etl/queue_spsc_atomic.h

```diff
@@ -365,6 +365,24 @@ namespace etl
     }
 #endif
 
+    //*************************************************************************
+    /// Peek the next value in the queue without removing it.
+    //*************************************************************************
+    bool peek(reference value)
+    {
+      size_type read_index = read.load(etl::memory_order_relaxed);
+
+      if (read_index == write.load(etl::memory_order_acquire))
+      {
+        // Queue is empty
+        return false;
+      }
+
+      value = p_buffer[read_index];
```

Comment:
I'm not a C++ expert by a long shot. I believe we do _not_ want to use move semantics here as we are not destroying the read object after. Is that correct?

Reply:
I think the function should be named 'front' to keep with the convention used by the STL, which the ETL follows.

Reply:
It's correct that it should not use move semantics, as it would invalidate the value for the subsequent accesses.

Reply:
I'll rename from peek to front. One follow on question to that, std::queue<T,Container>::front returns a reference to value in the queue rather than returning by value. Would we want to do a return by value like I have it or consider following the stdlib's method of return by reference (and I suppose we'd want a const overload version as well). What are your thoughts on that?

Reply:
I think it should follow the STL/ETL convention of returning a reference. It can be handy when the objects in the queue are large and a copy just to peek a value would be wasteful.

Reply:
Sounds good. I'll modify to match that and post a comment when I have it fixed up.